verticapy/tests_new/pipeline/test_validate.py

```diff
@@ -0,0 +1,124 @@
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+import pytest
+
+from verticapy import drop
+from verticapy._utils._sql._sys import _executeSQL
+from verticapy.datasets import load_winequality
+
+from verticapy.pipeline._validate import testing
+from verticapy.pipeline._train import training
+
+from verticapy.tests_new.pipeline.conftest import pipeline_exists
+
+
+class TestValidate:
+    """
+    Analytic Functions test
+    -
+    """
+
+    @pytest.mark.parametrize(
+        "test",
+        [
+            {
+                "metric": {
+                    "name": "aic",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "bic",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "max_error",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "mean_absolute_error",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "mean_squared_error",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "mean_squared_log_error",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+            {
+                "metric": {
+                    "name": "r2",
+                    "y_true": "quality",
+                    "y_score": "prediction",
+                }
+            },
+        ],
+    )
+    def test_regression(
+        self,
+        test,
+    ):
+        pipeline_name = "test_pipeline"
+        _executeSQL(f"CALL drop_pipeline('public', '{pipeline_name}');")
+
+        # Model Setup
+        table = load_winequality()
+        kwargs = {
+            "method": {
+                "name": "LinearRegression",
+                "target": "quality",
+                "params": {
+                    "tol": 1e-6,
+                    "max_iter": 100,
+                    "solver": "newton",
+                    "fit_intercept": True,
+                },
+            }
+        }
+
+        cols = ["fixed_acidity", "volatile_acidity", "citric_acid", "residual_sugar"]
+
+        # Part 1: Train a Model
+        _, model, _ = training(kwargs, table, pipeline_name, cols)
+
+        # Part 2: Run the Metrics
+        testing(test, model, pipeline_name, cols)
+
+        assert model
```

Comment:
but you are not saving the result of testing function? You may want to get the results of testing and then compare it with some values that we know prior. 

You could also check the content of the string output which could check the SQL to replicate the steps of the yaml file.

Reply:
The sql for testing looks like:
```sql
DROP TABLE IF EXISTS test_pipeline_METRIC_TABLE;CREATE TABLE test_pipeline_METRIC_TABLE(metric_name Varchar(100), metric FLOAT);
INSERT INTO test_pipeline_METRIC_TABLE
                SELECT 'mean_squared_log_error', subquery."AVG" FROM (
        SELECT AVG(POW(LOG(quality + 1) - LOG(prediction + 1), 2))
        FROM test_pipeline_PREDICT_VIEW
        WHERE quality IS NOT NULL
          AND prediction IS NOT NULL) as subquery;
COMMIT;
```

So I was thinking of something along these line:
```python
        assert 'DROP TABLE' in metric_sql
        assert 'CREATE TABLE' in metric_sql
        assert f'{pipeline_name}_PREDICT_VIEW' in metric_sql
        assert 'COMMIT;' in metric_sql
```
and some testing for the results
```python
        res = _executeSQL(f'SELECT * FROM {pipeline_name}_METRIC_TABLE;', method='fetchrow')

        _executeSQL(metric_sql)
        res_from_sql = _executeSQL(f'SELECT * FROM {pipeline_name}_METRIC_TABLE;', method='fetchrow')

        assert res == res_from_sql
        assert name in res
```

This way it will test:
1. On default pipeline_name_METRIC_TABLE is created correctly
2. The return script runs the exact same thing
3. The correct test is described in METRIC_TABLE
4. The main functions of the sql are included

Any thoughts?